.azure-devops/nightly.yml

```diff
@@ -100,38 +108,41 @@ stages:
 
       - template: templates/install-and-record-version.yml
 
-  - stage: 'test_min'
-    displayName: 'Run all tests with min CLI'
-    pool:
-      vmImage: ${{ parameters.vmImage }}
-    dependsOn: build
-    jobs:
-    - template: templates/nightly-tests.yml
-      parameters:
-        azureCLIVersion: min
-        pythonVersion: ${{ parameters.minCLIPythonVersion }}
-        testADT: ${{ parameters.testADT }}
-        testCentral: ${{ parameters.testCentral }}
-        testDPS: ${{ parameters.testDPS }}
-        testHub: ${{ parameters.testHub }}
-
-  - stage: 'test_msi'
-    displayName: 'Run all tests against Windows MSI install'
-    pool:
-      vmImage: 'windows-2019'
-    dependsOn: test_min
-    jobs:
-    - template: templates/nightly-tests.yml
-      parameters:
-        azureCLIVersion: msi
-        testADT: ${{ parameters.testADT }}
-        testCentral: ${{ parameters.testCentral }}
-        testDPS: ${{ parameters.testDPS }}
-        testHub: ${{ parameters.testHub }}
+  - ${{ each python_version in parameters.minCLIPythonVersions }}:
+    - stage: 'test_min_${{python_version}}'
+      displayName: 'Run all tests with min CLI on python ${{python_version}}'
+      pool:
+        vmImage: ${{ parameters.vmImage }}
+      dependsOn: build
+      jobs:
+      - template: templates/trigger-tests.yml
+        parameters:
+          azureCLIVersion: min
+          pythonVersion: ${{replace(python_version, '_', '.')}}
+          testADT: ${{ parameters.testADT }}
+          testCentral: ${{ parameters.testCentral }}
+          testDPS: ${{ parameters.testDPS }}
+          testHub: ${{ parameters.testHub }}
+
+  - ${{ each python_version in parameters.msiPythonVersions }}:
+    - stage: 'test_msi_${{python_version}}'
+      displayName: 'Run all tests against Windows MSI install on python ${{python_version}}'
+      pool:
+        vmImage: 'windows-2019'
+      dependsOn: test_min_${{python_version}}
+      jobs:
+      - template: templates/trigger-tests.yml
+        parameters:
+          azureCLIVersion: msi
+          pythonVersion: ${{replace(python_version, '_', '.')}}
```

Comment:
the MSI only has one python version

Reply:
Which python version is supported by MSI?

Reply:
It changes over time. The python runtime is loaded in the MSI, to be independent from system python runtimes. I believe more recently it provides a 3.8.x python version. In the past for a significant period its been 3.6.x. We should use the MSI python environment for tests - this is a key aspect for the MSI test.